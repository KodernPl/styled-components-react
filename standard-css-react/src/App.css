
@import './styles/variables.css';

.app {
  width: 100vw;
  height: 100vh;
  min-height: 56rem;
  padding-top: 3rem;
  display: flex;
  flex-direction: column;
  align-items: center;

  &--light {
    background: var(--color-sc-orange);
  }

  &--dark {
    background: linear-gradient(to top right, var(--color-gradient1), var(--color-gradient2), var(--color-gradient3), var(--color-gradient4));
  }

  @media (--desktop) {
    padding-top: 6rem;
  }
}
