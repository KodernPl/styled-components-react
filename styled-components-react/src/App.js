import React, { Component, Fragment } from 'react';
import styled, { ThemeProvider, css } from 'styled-components';
import styledTheming from 'styled-theming';

import Card from './components/Card';
import ThemeSwitcher from './components/ThemeSwitcher';
import Global from './styles/global';
import Normalize from './styles/normalize';
import { THEMES, COLORS, BREAKPOINTS } from './styles/variables';

const themeProps = styledTheming('mode', {
  [THEMES.light]: css`
    background: ${COLORS.scOrange};
  `,
  [THEMES.dark]: css`
  background: linear-gradient(to top right, ${COLORS.gradient1}, ${COLORS.gradient2}, ${COLORS.gradient3}, ${COLORS.gradient4});
`,
});

const CustomFlex = styled.div`
  ${themeProps};
  width: 100vw;
  height: 100vh;
  min-height: 56rem;
  padding-top: 3rem;
  display: flex;
  flex-direction: column;
  align-items: center;

  @media ${BREAKPOINTS.desktop} {
    padding-top: 6rem;
  }
`;

class App extends Component {
  state = {
    theme: THEMES.light,
  }

  handleThemeChange = () => {
    this.setState(prevState => ({
      theme: prevState.theme === THEMES.dark ? THEMES.light : THEMES.dark,
    }));
  }

  render() {
    const { theme } = this.state;

    return (
      <Fragment>
        <Normalize />
        <Global />
        <ThemeProvider theme={{ mode: theme }}>
          <CustomFlex>
            <Card />
            <ThemeSwitcher onThemeChange={this.handleThemeChange} theme={theme} />
          </CustomFlex>
        </ThemeProvider>
      </Fragment>
    );
  }
}

export default App;
