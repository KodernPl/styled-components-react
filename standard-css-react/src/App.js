import React, { Component } from 'react';
import cx from 'classnames';

import Card from './components/Card';
import ThemeSwitcher from './components/ThemeSwitcher';

import { ThemeProvider } from './utils/themeContext';
import { THEMES } from './styles/variables';

import './styles/normalize.css';
import './styles/global.css';
import './App.css';

const displayName = 'App';
const ns = 'app';

class App extends Component {
  state = {
    theme: THEMES.light,
  }

  handleThemeChange = () => {
    this.setState(prevState => ({
      theme: prevState.theme === THEMES.dark ? THEMES.light : THEMES.dark,
    }));
  }

  render() {
    const { theme } = this.state;
    const classNames = cx(
      ns,
      {
        [`${ns}--light`]: theme === THEMES.light,
        [`${ns}--dark`]: theme === THEMES.dark,
      },
    );

    return (
      <ThemeProvider value={theme}>
        <div className={classNames}>
          <Card />
          <ThemeSwitcher onThemeChange={this.handleThemeChange} theme={theme} />
        </div>
      </ThemeProvider>
    );
  }
}

App.displayName = displayName;

export default App;
