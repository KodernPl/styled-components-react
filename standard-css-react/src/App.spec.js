import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

const renderComponent = () => shallow(
  <App />,
);

test('renders without crashing', () => {
  expect(renderComponent()).toMatchSnapshot();
});
