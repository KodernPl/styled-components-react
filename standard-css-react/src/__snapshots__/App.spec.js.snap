// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<ContextProvider
  value="light"
>
  <div
    className="app app--light"
  >
    <Component />
    <ThemeSwitcher
      onThemeChange={[Function]}
      theme="light"
    />
  </div>
</ContextProvider>
`;
